src/Impostor.Api/Innersloth/Customization/PlayerOutfit.cs

```diff
@@ -2,19 +2,67 @@ namespace Impostor.Api.Innersloth.Customization
 {
     public class PlayerOutfit
     {
-        public ColorType Color { get; internal set; } = (ColorType)(-1);
+        private string _playerName = string.Empty;
+        private ColorType _color = (ColorType)(-1);
+        private string _hatId = "missing";
+        private string _petId = "missing";
+        private string _skinId = "missing";
+        private string _visorId = "missing";
+        private string _namePlateId = "missing";
 
-        public string HatId { get; internal set; } = "missing";
+        public bool IsDirty { get; internal set; }
 
-        public string PetId { get; internal set; } = "missing";
+        public string PlayerName
+        {
+            get => _playerName;
+            set => SetField(ref _playerName, value);
+        }
+
+        public ColorType Color
+        {
+            get => _color;
+            set => SetField(ref _color, value);
+        }
+
+        public string HatId
+        {
+            get => _hatId;
+            set => SetField(ref _hatId, value);
+        }
+
+        public string PetId
+        {
+            get => _petId;
+            set => SetField(ref _petId, value);
+        }
+
+        public string SkinId
+        {
+            get => _skinId;
+            set => SetField(ref _skinId, value);
+        }
+
+        public string VisorId
+        {
+            get => _visorId;
+            set => SetField(ref _visorId, value);
+        }
+
+        public string NamePlateId
+        {
+            get => _namePlateId;
+            set => SetField(ref _namePlateId, value);
+        }
+
+        public byte HatSequenceId { get; internal set; } = 0;
 
-        public string SkinId { get; internal set; } = "missing";
+        public byte PetSequenceId { get; internal set; } = 0;
 
-        public string VisorId { get; internal set; } = "missing";
+        public byte SkinSequenceId { get; internal set; } = 0;
 
-        public string NamePlateId { get; internal set; } = "missing";
+        public byte VisorSequenceId { get; internal set; } = 0;
 
-        public string PlayerName { get; internal set; } = "missing";
+        public byte NamePlateSequenceId { get; internal set; } = 0;
```

Comment:
Should we be exposing sequence ids in the public api?

Reply:
Probably not, although we should add an option to set them using Impostor.Server.

Reply:
I've made them private for now